doc/release-notes-1.2.0.rst

```diff
@@ -0,0 +1,502 @@
+    This library was extracted from the :ref:`bootloader` sample.
+
+Updated samples and applications
+--------------------------------
+
+* :ref:`bootloader`:
+
+  * Moved the provisioning data (slot sizes/addresses and public keys) to one-time programmable (OTP) memory for nRF9160 devices.
+  * Implemented invalidation of public keys.
+
+
+Updated libraries
+-----------------
+
+* :ref:`doc_fw_info`:
+
+  * Renamed ABIs to EXT_APIs.
+  * Restructured the :c:type:`fw_info` structure:
+
+    * Renamed the fields ``firmware_size``, ``firmware_address``, and ``firmware_version`` to ``size``, ``address``, and ``version``.
+    * Added a field to invalidate the structure.
+    * Added reserved fields for future use.
+    * EXT_APIs are now in a list at the end of the structure, instead of being available behind a function call.
+    * EXT_APIs can now be requested by adding a request structure to a list after the EXT_API list itself.
+
+  * Updated how EXT_API requests are checked.
+    Requests are now checked against EXT_APIs by the bootloader before booting the image.
+  * Added two new allowed offsets for the struct: 0x0 and 0x1000 bytes.
+  * Removed ``memeq()`` in favor of regular ``memcmp()``.
+  * Renamed ``__ext_api()`` to ``EXT_API()``, because names starting with ``__`` are reserved for the compiler.
+  * Added new configuration options ``CONFIG_*_EXT_API_REQUIRED`` and ``CONFIG_*_EXT_API_ENABLED`` for, respectively, users and providers of EXT_APIs.
+
+* :ref:`lib_dfu_target`:
+
+  * Added the configuration option :option:`CONFIG_DFU_TARGET_MCUBOOT_SAVE_PROGRESS`, which uses Zephyr's :ref:`zephyr:settings` subsystem.
+    When this option is enabled, the write progress of an MCUboot style upgrade is stored, so that the progress is retained when the device reboots.
+  * Fixed a bug where :cpp:func:`dfu_target_done` printed ``unable to deinitialize dfu resource`` when no target was initialized. ??? did it actually print it or return an error?
+
+* Moved the following libraries from ``drivers/`` to ``lib/``:
+
+  * :ref:`fprotect_readme`
+  * :ref:`st25r3911b_nfc_readme`
+  * ``adp536x``
+  * ``flash_patch``
+
+
+Crypto
+======
+
+* Added low-level cryptographic test suite using NIST, RFCs, and custom test vectors.
+* :ref:`nrf_cc310_mbedcrypto_readme`/:ref:`nrf_cc310_platform_readme` v0.9.2:
+
+  * Fixed power-efficiency issues.
+  * Added experimental use of CryptoCell interrupt instead of busy-waits.
+* :ref:`lib_hw_cc310`:
+
+  * Added support for CryptoCell interrupt.
+
+nRF BLE Controller
+==================
+
+* Updated the :ref:`nrfxlib:ble_controller` to v. ??? what version? The changelog seems to be messed up.
+  For details, see the :ref:`nrfxlib:ble_controller_changelog`. ??? update link
+
+Multi-Protocol Service Layer (MPSL)
+===================================
+
+* Added MPSL v0.5.0-1.prealpha, a library of common services for single and multi-protocol implementations.
+
+
+Subsystems
+==========
+
```

Comment:
MPSL has been added as a new subsystem. I'm not sure if it should be mentioned here
```suggestion
Multi-Protocol Service Layer (MPSL)
---------------------------------------
 * MPSL has been added as a new subsystem. It integrates the MPSL into the NCS environment

```

Reply:
Agree this should be included. I've also mention this in the Highlight.